week2/wordle.c

```diff
@@ -0,0 +1,220 @@
+            char options[LISTSIZE][wordsize + 1];
+
+            for (int i = 0; i < LISTSIZE; i++)
+            {
+                fscanf(wordlist, "%s", options[i]);
+            }
+
+            // pseudorandomly select a word for this game
+            srand(time(NULL));
+            string choice = options[rand() % LISTSIZE];
+
+            // allow one more guess than the length of the word
+            int guesses = wordsize + 1;
+            bool won = false;
+
+            // print greeting, using ANSI color codes to demonstrate
+            printf(GREEN"This is WORDLE50"RESET"\n");
+            printf("You have %i tries to guess the %i-letter word I'm thinking of\n", guesses, wordsize);
+
+            // main game loop, one iteration for each guess
+            for (int i = 0; i < guesses; i++)
+            {
+                // obtain user's guess
+                string guess = get_guess(wordsize);
+
+                // array to hold guess status, initially set to zero
+                int status[wordsize];
+
+                // set all elements of status array initially to 0, aka WRONG
+                // TODO #4
+                for (int status_element = 0; status_element <= wordsize; status_element++)
+                {
+                    status[status_element] = 0;
+                }
+
+                // Calculate score for the guess
+                int score = check_word(guess, wordsize, status, choice);
+
+                printf("Guess %i: ", i + 1);
+
+                // Print the guess
+                print_word(guess, wordsize, status);
+
+                // if they guessed it exactly right, set terminate loop
+                if (score == EXACT * wordsize)
+                {
+                    won = true;
+                    break;
+                }
+            }
+
+            // Print the game's result
+            // TODO #7
+
+            if (won == true)
+            {
+                printf("You've won!\n");
+                return 0;
+            } else 
+            {
+                printf("The word was: %s \n", choice);
+            }
+
+            // that's all folks!
+        } 
+        else 
+        {
+           printf("Error: wordsize must be either 5, 6, 7 or 8 \n"); 
+           return 1;
+        }                      
+    } 
+    else 
+    {
+        printf("Usage: ./wordle wordsize \n");
+        return 1;
+    }
+}
+
+string get_guess(int wordsize)
+{
+    // ensure users actually provide a guess that is the correct length
+    // TODO #3
+
+    string guess = get_string ("Input a %i-letter word: ", wordsize);
+
+    if (strlen(guess) == wordsize)
+    {
+        return guess;
+    } 
+    else
+    {
+        while (strlen(guess) != wordsize)
+        {
+            string guess = get_string ("Input a %i-letter word: ", wordsize);
+
+            if (strlen(guess) == wordsize)
+            {
+                return guess;
+                break;
```

Comment:
```suggestion
                return guess;
```
Wenn du vorher returnest, wird nach dem return statement nichts mehr durchlaufen.